js/aax.js

```diff
@@ -1902,6 +1903,95 @@ module.exports = class aax extends Exchange {
         return this.parseDepositAddress (data, currency);
     }
 
+    async fetchDeposits (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {
+            // status Not required -  Deposit status, "1: pending,2: confirmed, 3:failed"
+            // currency: Not required -  String Currency
+            // startTime Not required Integer Default: 90 days from current timestamp.
+            // endTime Not required Integer Default: present timestamp.
+        };
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+            request['currency'] = currency['id'];
+        }
+        if (since !== undefined) {
+            request['startTime'] = since; // default 90 days
+        }
+        const response = await this.privateGetAccountDeposits (this.extend (request, params));
+        // {    "code": 1,
+        //     "data": [{
+        //         "currency": "USDT",
+        //         "network": "USDT",
+        //         "quantity": "19.000000000000",
+        //         "txHash": "75eb2e5f037b025c535664c49a0f7cc8f601dae218a5f4fe82290ff652c43f3d",
+        //         "address": "1GkB7Taf7uttcguKEb2DmmyRTnihskJ9Le",
+        //         "status": "2",
+        //         "createdTime": "2021-01-08T19:45:01.354Z",
+        //         "updatedTime": "2021-01-08T20:03:05.000Z",
+        //     }]
+        //     "message": "success",
+        //     "ts": 1573561743499
+        // }
+        const deposits = this.safeValue (response, 'data', []);
+        return this.parseTransactions (deposits, code, since, limit);
+    }
+
+    parseTransactionStatus (status) {
+        const statuses = {
+            '1': 'pending',
+            '2': 'ok',
+            '3': 'failed',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        //    {
+        //         "currency": "USDT",
+        //         "network": "USDT",
+        //         "quantity": "19.000000000000",
+        //         "txHash": "75eb2e5f037b025c535664c49a0f7cc8f601dae218a5f4fe82290ff652c43f3d",
+        //         "address": "1GkB7Taf7uttcguKEb2DmmyRTnihskJ9Le",
```

Comment:
A quick question – is it the destination address? If it is, we should probably also fill the `addressTo` field below. If it's the source address, we should fill `addressFrom` as well. Most often the exchanges will post the destination address there, but we have to make sure that we treat it correctly in each specific case.

Reply:
One more thing, I think they will also add `addressTag` field for blockchains that require address + tag for wallet identification. Despite that it is missing in this specific commented deposit sample, other txs may have more fields. We should probably add that as `tag`. And also as either `tagFrom` or `tagTo`.

Reply:
Good question. I'm not entirely sure, but from the docs I'd assume it's the address to which the funds are deposited to.
https://apidoc.cn.atomintl.com/#get-deposit-history

Reply:
For the `tag` it is returned in the [Query Deposit Address](https://www.aax.com/apidoc/index.html?python#query-deposit-address)
Are you saying we should call also that endpoint to the get the `tag` of the addresses of each currency? Or you think the docs of aax may be outdatted and sometimes they return a `tag` field in the getDepositHistory?

Reply:
Nope, it's related, but it's not exactly what i'm trying to say.
First, there's two types of blockchains, so to speak:
1. blockchains that require just the address where to send the transaction to (BTC, ETH, and alike)
2. blockchains that require address + tag (like XRP, XLM, the tag is used for additional identification and routing, the tag is also often called memo or payment_id)

Most of the exchanges will have the support for both types of blockchains.
So, when, depositing to this exchange, you could deposit funds through a blockchain of type 2, with a tag.
If you go to their website account and try depositing XLM – you will see this:

<img width="1459" alt="Screenshot 2022-01-08 at 14 50 48" src="https://user-images.githubusercontent.com/1294454/148643044-c93a0b11-2122-4df9-9607-bee81c2a3a35.png">

So, if we deposit XLM to AAX, then our deposit would obtain a tag, apart from the address. And the `addressTag` field would appear in our deposit history in the corresponding deposit structure. They don't mention this in their API docs, but there's a tiny hint – you can look at the withdrawal structure in their docs – that one has an additional `addressTag` field.

So, we can assume, that the `addressTag` field will appear on some deposits. If that is the case, we should take their `addressTag` and convert it to our `tag`. And since the address also can be a destination address or a source address, the tag can also be a destination tag or a source tag, so we should will `tag` + `tagFrom` or `tag` + `tagTo`.

Lmk if that is still not very clear.

Reply:
Got it! Thanks for the clear explanation! 
The strange thing is they use `addressTag`  in the structure of `/account/withdraws` endpoint but then `tag` in the `/account/deposit/address` endpoint.
I'll see if I can find out with the tests requests which one they use for deposits, but as a PlanB I think we can always check if any of the both properties are defined.

Reply:
> but as a PlanB I think we can always check if any of the both properties are defined.

Yes, we just add the following to the parseTransaction – it will work in all cases:

```JavaScript
parseTransaction (transaction, currency = undefined) {
    // ...
    const tag = this.safeString (transaction, 'addressTag');
    // ...
    return {
        // ...
        'tagTo': tag, // or change places with tagFrom, depending if it's a destination tag or a source tag
        'tag': tag,
        'tagFrom': undefined, // or vice versa with tagTo, depending if it's a destination tag or a source tag
        // ...
    };
}
```

Reply:
I created an account on AAX, did an XRP deposit and tested the API using the cli.js. And learnt two things:
- The `address` is actually the `addressFrom`
- There is no `tag` or `addressTag` in the response, even though it was an XRP deposit.

I made the corresponding changes in the last commit. And for the tag I left this code to future proof it: 
```
const tag = this.safeString2 (transaction, 'tag', 'addressTag');
```

Reply:
@pcriadoperez great! Thank you very much!